javascript/commons/FilterButtons.js

```diff
@@ -211,6 +233,58 @@ liquipedia.filterButtons = {
 				}
 			} );
 		} );
+
+		this.templateExpansions.forEach( ( templateExpansion ) => {
+			const isDefault = templateExpansion.groups.every( ( group ) => {
+				const filterGroup = this.filterGroups[ group ];
+				if ( filterGroup.curated || filterGroup.defaultCurated ) {
+					return filterGroup.curated === filterGroup.defaultCurated;
+				}
+				return Object.keys( filterGroup.filterStates ).every( ( filterState ) => {
+					return filterGroup.filterStates[ filterState ] === filterGroup.defaultStates[ filterState ];
+				} );
+			} );
+			if ( isDefault ) {
+				templateExpansion.element.innerHTML = templateExpansion.cache.default;
+				return;
+			}
+			const parameters = templateExpansion.groups.map( ( group ) => {
+				if ( this.filterGroups[ group ].curated ) {
+					return group + '=curated';
+				}
+
+				const filterStates = this.filterGroups[ group ].filterStates;
+				const activeFilters = Object.keys( filterStates ).filter( ( k ) => filterStates[ k ] );
+
+				return group + '=' + activeFilters.toString();
+			} );
+			const wikitext = '{{' + templateExpansion.template + '|' + parameters.join( '|' ) + '}}';
+
+			if ( wikitext in templateExpansion.cache ) {
```

Comment:
Could reduce cache key length by using only the content of `parameters`

Reply:
I think using the wikitext variable is ok, I wouldn't be worried about the eĺength of the cache key, it only exists in memory and is unlikely to be of a size that will eat up someones entire RAM